### PR TITLE
Add crust-gather to krew plugins

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
This changes adds [crust-gather](https://github.com/crust-gather/crust-gather) to the list of supported plugins.

Crust-gather is a debugging tool, which allows user to collect full or partial cluster state and store it in the filesystem. Later, the stored directory can be served via api-server replica, to analyze via kubectl or k9s. This includes:
- pod logs
- node kubelet logs
- all custom and in-tree resources

Crust-gather allows to specify filters on the resource selection, so only some resources can be collected.

If there is a need to exclude secret values from the output, there are flags to pass either an env variable with a secret or a [secrets file](https://github.com/crust-gather/crust-gather/pull/4) with a list of secrets.

To view all possible configurations after installation: `kubectl crust-gather --help`

Here is a demo of the functionality:
[asciicast](https://asciinema.org/a/632848)

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

File: plugins/crust-gather.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        os: darwin
+        arch: arm64
+  - bin: kubectl-crust-gather
+    uri: https://github.com/crust-gather/crust-gather/releases/download/v0.5.3/kubectl-crust-gather_v0.5.3_darwin_amd64.tar.gz
+    sha256: 506fb5be53170cb2f9edb0842b616aed8e76c8b832f61d9e46ec2f636d880500
+    selector:
+      matchLabels:
+        os: darwin
+        arch: amd64
+  shortDescription: Provides functionality similar to os adm inspect, by collecting all resources for debugging purposes, and serving them via simple api-server.
```
Comment (ahmetb) [inline plugins/crust-gather.yaml:45]: this won't fit and will be always truncated. limit to <60 chars.
https://krew.sigs.k8s.io/docs/developer-guide/plugin-manifest/

File: plugins/crust-gather.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  description: |
+    Provides functionality similar to os adm inspect, by collecting all resources for debugging purposes. Later, the archive can be served
+    with an api-server to explore the content using regular clients, like kubectl or k9s.
```
Comment (ahmetb) [inline plugins/crust-gather.yaml:48]: I wouldn't mention "os adm inspect" here. Forget about that and approach writing this from this perspective:  I'm someone who's looking at this tool for the first time, what does it do for me. Focus on that.


Also consider line wrapping this at 80 chars please.

File: plugins/crust-gather.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  version: v0.5.3
+  homepage: https://github.com/crust-gather/crust-gather
+  platforms:
+  - bin: kubectl-crust-gather.exe
+    uri: https://github.com/crust-gather/crust-gather/releases/download/v0.5.3/kubectl-crust-gather_v0.5.3_windows_amd64.tar.gz
+    sha256: 6c3e3697ad33ad190fd2398776fc9a2b28288147c57991878c9d48ed38063f23
+    selector:
+      matchLabels:
+        os: windows
+        arch: amd64
```
Comment (ahmetb) [inline plugins/crust-gather.yaml:16]: consider releasing for windows/arm64 later (doesn't have to be in this PR).

File: plugins/crust-gather.yaml (hunk 1 above)
Comment (Danil-Grigorev) [reply plugins/crust-gather.yaml]: Should be fixed.

File: plugins/crust-gather.yaml (hunk 2 above)
Comment (Danil-Grigorev) [reply plugins/crust-gather.yaml]: Updated. Makes sense, the description is very much outdated, as the plugin works on a different pattern.

File: plugins/crust-gather.yaml (hunk 3 above)
Comment (Danil-Grigorev) [reply plugins/crust-gather.yaml:16]: I’ll have a look into this, thanks

File: plugins/crust-gather.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        os: darwin
+        arch: arm64
+  - bin: kubectl-crust-gather
+    uri: https://github.com/crust-gather/crust-gather/releases/download/v0.5.3/kubectl-crust-gather_v0.5.3_darwin_amd64.tar.gz
+    sha256: 506fb5be53170cb2f9edb0842b616aed8e76c8b832f61d9e46ec2f636d880500
+    selector:
+      matchLabels:
+        os: darwin
+        arch: amd64
+  shortDescription: Plugin to collect cluster state and serve it via an api server
```
Comment (ahmetb) [inline plugins/crust-gather.yaml:45]: Sorry for pestering you but everything is a plugin in Krew so it's obvious :) 
```suggestion
  shortDescription: Collect cluster state and serve it via an api server
```

Comment (Danil-Grigorev) [reply plugins/crust-gather.yaml]: Np, I’ve updated the description to give more details on that front, and simplified the short one :)

Comment (ahmetb) [reply plugins/crust-gather.yaml]: Typically usage strings in description are not ideal but you can fix in a follow up.

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!
Also, please consider setting up [Krew release automation](https://krew.sigs.k8s.io/docs/developer-guide/release/automating-updates/) which helps you skip manually making updates to your Krew manifests on each new version and send a pull request. It’s a GitHub workflow bot that can run every time you push a new tag. Those PRs are automatically approved.

**Pull request merged**